### PR TITLE
Daily work summary refactor

### LINKED ISSUE #12376: Daily Work Summary Email
Currently, What did you work Today? email is going to active Employees only. It should go to active employees and users too.

Use case: We have some foundation employee added as a User in frappe.erpnext.com. But to have daily time log triggered to them, will also need to add them as Employee. Will be better if Daily timelog summary is emailed based on Active Users, and not based on Employee.

### PR DESCRIPTION
**Pull request by surajshetty3416**
Fixed https://github.com/frappe/erpnext/issues/12376

Now we can have multiple settings for Daily Work Summary which can have different set of users with separate _holiday list_ and _time to send emails_

### CODE REVIEW

File: erpnext/docs/user/manual/en/human-resources/daily-work-summary.md [hunk 1]
```diff
@@ -1,15 +1,24 @@
...
-Go to "Daily Work Summary Settings" via HR module or search bar and set the company for which you want to activate this feature.
+Go to "Daily Work Summary Setting" via HR module or search bar and set the users for whom you want to send the reminder.
+
+You can set multiple settings with different set of _users_ from your user list with different _time to send emails_ and with seperate _holiday list_ for each.
+
+You can also choose to customize the _Message_ you send to users.
 
-You can also choose to Customize the Message you send to your employees:
+**Note:**
+>1. If no holiday list is seleted then the email will be sent every day.
```
Comment (netchampfaris) [inline erpnext/docs/user/manual/en/human-resources/daily-work-summary.md]: selected

File: erpnext/docs/user/manual/en/human-resources/daily-work-summary.md [hunk 2]
```diff
@@ -1,15 +1,24 @@
...
+>2. Setting name will be sent as the title of daily summary email.
+>3. Deactivated settings will be skipped from daily check.
```
Comment (netchampfaris) [inline erpnext/docs/user/manual/en/human-resources/daily-work-summary.md]: What does this mean?

**Comment by olamide2** [comment]
Why not some screen shots to allow us appreciate the changes ?

**Review by netchampfaris — DISMISSED** [review]

File: erpnext/docs/user/manual/en/human-resources/daily-work-summary.md (hunk 1 above)
Comment (surajshetty3416) [reply erpnext/docs/user/manual/en/human-resources/daily-work-summary.md]: https://github.com/surajshetty3416/erpnext/commit/102cd445a34e0c1116ec5576a4fd0592a6297f5f

File: erpnext/docs/user/manual/en/human-resources/daily-work-summary.md (hunk 2 above)
Comment (surajshetty3416) [reply erpnext/docs/user/manual/en/human-resources/daily-work-summary.md]: https://github.com/surajshetty3416/erpnext/commit/102cd445a34e0c1116ec5576a4fd0592a6297f5f

File: erpnext/hr/doctype/daily_work_summary_group/daily_work_summary_group.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+from frappe.model.document import Document
+import frappe.utils
+from frappe import _
+
+
+class DailyWorkSummaryGroup(Document):
+	def validate(self):
+		if self.users:
+			if not frappe.flags.in_test and not is_incoming_account_enabled():
+				frappe.throw(_('There must be a default incoming Email Account enabled for this to work. '
```
Comment (rmehta) [inline erpnext/hr/doctype/daily_work_summary_group/daily_work_summary_group.py]: translatable strings must be in single line. Use smaller strings

Maybe just _('Please enable default incoming account before creating Daily Work Summary Group')

Comment (surajshetty3416) [reply erpnext/hr/doctype/daily_work_summary_group/daily_work_summary_group.py]: Fixed https://github.com/frappe/erpnext/pull/12944/commits/343192078b2fffb4a4b524731713d422e7f9dd89

**Pull request merged**